### DISCUSSION TITLE
我试图用v2ray作为代理然后远程联机，有办法吗

### DISCUSSION THREAD

**Discussion by sunianOvO** [v2fly/v2ray-core#2372] [Q&A]
目标效果:

设备A在北京 ，设备B在广州 两台设备连接同一个vmess节点，然后 设备A 打开一个局域网联机游戏，设备B 可以加入到 设备A 的对局中

有人知道应该怎么配置吗

**Comment by mydogshitgold** [comment]
Do both devices have IPv6 addresses? Most ISPs now provide  IPv6 addresses. Unlike IPv4 behind NAT, IPv6 is end-to-end. You can simply connect two hosts using IPv6 directly without the complexity of V2ray and potentially have lower latency than using a intermediate relay

Use `ipconfig` on Windows, `ip addr` on linux and `ipconfig getifaddr <interface name>` on Mac to find your IPv6 address. Alternatively you can also do: `dig +short myip.opendns.com @resolver1.opendns.com AAAA`

**Comment by nursery01** [comment]
可以在client.json裏面關閉本地路由
v2ray server run game [::1]:123
A [::1]:123
B [::1]:123
就是你要在伺服器裏運行游戲并且開啓房間

```
             |------------v2ray 123.123.123.123:300
server ------|
             |-------------game serivce [::1]:123
```

```
client A(off v2 route) -----------[::1]:123
client B(off v2 route) -----------[::1]:123
```

**Reply by mydogshitgold** [reply]
This should be avoided.

Both client and server are exposed to high security risks, in addition to confusion, when loopback addresses are proxied,

**Reply by nursery01** [reply]
yes i was wrong
SSRF (Server-Side Request Forgery) security bug
but, i set route. [::1]:4**** will also be sent to v2 server
ipv4 is seems normal
so, v2 have security bug?

**Reply by mydogshitgold** [reply]
That's not what I mean.

In general, anything that exposes loopback addresses should be avoided. You can use other reserved address block e.g. `fc00::/7`

**Reply by sunianOvO** [reply]
这可能不是我想要的，或者我没看明白，我的希望是同一台服务器下的用户处于一个虚拟局域网，让服务器充当家里的路由器，连接节点相当于在家里链接WIFI

**Reply by mydogshitgold** [reply]
What you have just described is a [VPN](https://en.wikipedia.org/wiki/Virtual_private_network). V2Ray, at best, works like a [proxy](https://en.wikipedia.org/wiki/Proxy_server).

**Reply by mydogshitgold** [reply]
It is best to avoid any man-in-the-middle if you want to connect hosts together and play games (It's your original description) , or the latency is going to be ugly.

---

For example, you have a server in Tokyo:

Beijing <- **2101** km -> Tokyo <- **2910** km -> Guangzhou: `total 5011 km`
Beijing <- **1890** km -> Guangzhou: `total 1890 km`
Bear in mind that there will be hops between autonomous systems.

**Reply by nursery01** [reply]
用我描述的這種方法伺服器確實能成為一個路由器，A和B之間是能夠通過伺服器通信的。⚠️不過這種作法不安全⚠️
